src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4323,7 +4329,11 @@ LogicalResult ONNXScatterOp::inferShapes(
 
 LogicalResult ONNXScatterElementsOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  if (!data().getType().isa<mlir::RankedTensorType>())
+    return success();
+
+  getResult().setType(data().getType());
+  return success();
```

Comment:
Scatter Elements operation also has some requirements that are not tested at this time:
```
nputs

data (differentiable) : T
Tensor of rank r >= 1.
indices (non-differentiable) : Tind
Tensor of int32/int64 indices, of r >= 1 (same rank as input). All index values are expected to be within bounds [-s, s-1] along axis of size s. It is an error if any of the index values are out of bounds.
updates (differentiable) : T
Tensor of rank r >=1 (same rank and shape as indices)
```

Easiest is to add these constraints while doing the shape inference, and add them in a separate verifier